src/components/CreateVmWizard/steps/BasicSettings.js

```diff
@@ -518,12 +552,24 @@ class BasicSettings extends React.Component {
               />
             </FieldRow>
 
+            {/*  Configure Timezone checkbox */}
+            <FieldRow id={`${idPrefix}-sysPrepTimezone-configure`} vertical>
+              <Checkbox
+                id={`${idPrefix}-sysprep-timezone-config`}
+                checked={data.enableSysPrepTimezone}
+                onChange={e => this.handleChange('enableSysPrepTimezone', e.target.checked)}
+              >
+                {msg.sysPrepTimezoneConfigure()}
+              </Checkbox>
+            </FieldRow>
+
             <FieldRow label={msg.sysPrepTimezone()} id={`${idPrefix}-sysPrepTimezone`} vertical>
               <SelectBox
                 id={`${idPrefix}-sysprep-timezone-select`}
                 items={timezones}
-                selected={data.initTimezone}
+                selected={data.selectedSysprepTimezone}
```

Comment:
We use different variable to store selected and on-change values. Please explain (in a comment) why this approach is used and what is the relationship between those 2 values.

Reply:
This is intentional, it allows us to achieve the expected behavior as in Admin Portal - to see the _(GMT) Greenwich Standard Time_ selected in the drop down in case of no sysprep timezone set => these are two different values that we need at the same time. In this and some other cases, the selected/displayed sysprep timezone and the one which is actually set are not always the same.

This approach also allows us to remember the last selected sysprep timezone in cases when, for example, user disables sysprep completely or decides not to configure the sysprep timezone (and then they can change their mind, before creating a VM, of course). I was trying to copy the behavior in Admin Portal where we remember the last selected value, too, after disabling sysprep or only the sysprep timezone drop down.

I hope I made it more clear. However I think there are more possibilities how to implement such an approach.

Reply:
> Please explain (in a comment) 

Sorry for not being clear about this. I've meant a comment in the code :) 
Just to help others understand the design choice.

> achieve the expected behavior as in Admin Portal 

It seems Admin Portal uses different UI sections for sysprep (Windows-only) and for cloud init(other OS) whereas we merged them into one section. Interestingly in both sections Admin Portal allows to set the time zone.
In the end values from both sources are copied to `VmInit.timeZone`(see `VmInitModel.buildModelSpecificParameters`).
I would assume that it will work similar in the Vm Portal - output contains one value (`timeZone`), windows uses `sysprepTimeZone`, others use `cloudInitTimeZone`.

> in Admin Portal where we remember the last selected value,

Admin Portal stores that in the model. In React this could be implemented as uncontrolled component maintaining it's own state or by adding some draft/history state in the step. Such functionality is nice but if it would make code harder to understand I'm ready to sacrifice it :)

Reply:
Regarding your second sub-comment, as I properly understand, it's related to your suggestions of not using `timeZone` or `initTimezone` but to completely change all that timezone related stuff we already use it here for some time. Well, ok, but I think that's for another PR and another discussions.

Regarding your last sub-comment. I think it's not so much code and it already works well. Some little refactoring of the actual changes + more comments in the code would definitely improve it. But if we don't add this functionality, we make the users more confused because the same thing will work differently in VM Portal than in webadmin. And I think this functionality isn't anything complicated. I don't agree with the approach of not adding this functionality just because "it would make code harder to understand". We need this functionality to make our UIs more consistent. That's the purpose of this PR - consistency. This PR does not fix any bug regarding creating VMs and timezones.

Reply:
>  it's related to your suggestions of not using timeZone or initTimezone but to completely change all that timezone related stuff we already use it here for some time.

Not exactly. It's more about using the same names in the same context as in Admin Portal. Using `cloudInitTimeZone` as synonym for `sysprepTimeZone` might be confusing.

> I think it's not so much code and it already works well. 

OK

> That's the purpose of this PR - consistency

@sgratch  Perhaps we should add time zone support also for other OS? Admin Portal seems to support that.

Reply:
I was wondering why it is in webadmin and not also in VM Portal, too. I've just mentioned that in some other comment.

Reply:
@sgratch @sjd78 WDYT?

Reply:
Conclusion after some conversation with Sharon about this: "there was no customer requirement to support it for cloud-init as well. I would leave it as is for now."